tests/ztest/zexpect/src/main.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+#include <zephyr/ztest.h>
+
+ZTEST_SUITE(expect, NULL, NULL, NULL, NULL, NULL);
+
+ZTEST_EXPECT_FAIL(expect, test_fail_later);
+ZTEST(expect, test_fail_later)
+{
+	void *empty_ptr = NULL;
+	uint32_t val = 5;
+
+	zexpect_equal(val, 2);
+	zexpect_not_equal(val, 5);
+
+	zexpect_not_null(empty_ptr);
+
+	zassert_true(true);
+}
+
+ZTEST(expect, test_pass_expect_true)
+{
+	zexpect_true(true);
+}
+
+ZTEST_EXPECT_FAIL(expect, test_fail_expect_true);
+ZTEST(expect, test_fail_expect_true)
```

Comment:
could we wrapper ZTEST into ZTEST_EXPECT_FAIL? so that we do not need this two sequence statges call?

Reply:
We should make an issue and probably assign it to @yperess or me as the `ZTEST_EXPECT_FAIL/SKIP` macros are outside the scope of this PR.

Reply:
@yperess , comments?

Reply:
@hakehuang Can we make this a separate GitHub issue as this PR in no way implements the `ZTEST_EXPECT_FAIL` macro.

Reply:
ok

Reply:
Sorry, I somehow missed this. @hakehuang I'm not sure what you mean by "wrapping". If you file an issue I'll be happy to look at what you're proposing.